### PR TITLE
Re-add SipHash keyed functions

### LINKED ISSUE #50311: sipHash64Keyed not same sipHash64
Testing queries (query 6 return Error):

//------------------------- Запрос 1 -------------------------
DROP TABLE IF EXISTS sipHash64Keyed_test
;
//------------------------- Запрос 2 -------------------------
CREATE TABLE sipHash64Keyed_test
ENGINE = Memory() 
AS
SELECT
   1 a,
   'test' b
;
//------------------------- Запрос 3 -------------------------
SELECT
  *
FROM  
  sipHash64Keyed_test
FORMAT TSVWithNamesAndTypes
;
//------------------------- Запрос 4 -------------------------
SELECT
  sipHash64Keyed((toUInt64(0), toUInt64(0)), 1, 'test')
FORMAT TSVWithNamesAndTypes
;
//------------------------- Запрос 5 -------------------------
SELECT
  sipHash64(tuple(*))
FROM  
  sipHash64Keyed_test
FORMAT TSVWithNamesAndTypes
;
//------------------------- Запрос 6 -------------------------
SELECT
  sipHash64Keyed((toUInt64(0), toUInt64(0)), tuple(*))
FROM  
  sipHash64Keyed_test
FORMAT TSVWithNamesAndTypes
Error

### PR DESCRIPTION
**Pull request by aiven-sal**
This PR fix the known issues that the old implementation had.

Related to PRs #45513 #50426 #50245 and commit https://github.com/ClickHouse/ClickHouse/pull/51455/commits/3ed86867f8b2c055fab7903619f0a74650742722

Closes #50311

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Re-add SipHash keyed functions

### Documentation entry for user-facing changes

- [x] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/queries/0_stateless/02534_keyed_siphash.sql [hunk 1]
```diff
@@ -281,3 +281,74 @@ INSERT INTO tab VALUES ((2, 2), 4);
...
+sipHashKeyed_test;
+
+SELECT
+hex(sipHash128Keyed((toUInt64(0), toUInt64(0)), a, b))
+FROM
+sipHashKeyed_test;
+
+DROP TABLE sipHashKeyed_test;
+
+SELECT 'Check multiple keys';
```
[the remark this answers is not in the document]
Comment (rschu1ze) [reply tests/queries/0_stateless/02534_keyed_siphash.sql:331]: Thanks. In general, systematic tests for all variants always make sense, even if the implementation is shared.

File: src/Functions/FunctionsHashing.h [hunk 2]
```diff
@@ -1696,7 +1839,7 @@ struct NameIntHash32 { static constexpr auto name = "intHash32"; };
 struct NameIntHash64 { static constexpr auto name = "intHash64"; };
 
 using FunctionSipHash64 = FunctionAnyHash<SipHash64Impl>;
-using FunctionSipHash64Keyed = FunctionAnyHash<SipHash64KeyedImpl, true, SipHash64KeyedImpl::Key>;
+using FunctionSipHash64Keyed = FunctionAnyHash<SipHash64KeyedImpl, true, SipHash64KeyedImpl::Key, SipHash64KeyedImpl::KeyColumns>;
```
[the remark this answers is not in the document]
Comment (rschu1ze) [reply src/Functions/FunctionsHashing.h:1844]: SFINAE magic sounds not very desirable 😄
It is okay then ... I don't have a better idea at the moment.

**Review by rschu1ze — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
#52466

@aiven-sal sorry for this revert; it is just for the principle - quickly revert instead of letting errors in the CI.
Let's fix and add it again.

**Comment by alexey-milovidov** [comment]
@aiven-sal it makes sense to continue.

**Comment by alexey-milovidov** [comment]
@aiven-sal, do we still need keyed SipHash functions?

**Pull request merged**